pallets/parachain-staking/src/lib.rs

```diff
@@ -1375,6 +1390,73 @@ pub mod pallet {
 
 			Ok(().into())
 		}
+
+		/// Notify a collator is inactive during MaxOfflineRounds
+		#[pallet::call_index(29)]
+		#[pallet::weight(
+			// TODO: add the proper benchmark function once it is available
+			T::DbWeight::get().reads_writes(0 as u64, 0 as u64)
+		)]
+		pub fn notify_inactive_collator(
+			origin: OriginFor<T>,
+			collator: T::AccountId,
+		) -> DispatchResult {
+			ensure!(
+				<EnableMarkingOffline<T>>::get(),
+				<Error<T>>::MarkingOfflineNotEnabled
+			);
+			ensure_signed(origin)?;
+
+			let collators = <SelectedCandidates<T>>::get();
```

Comment:
Rather than read and decode the entire `<SelectedCandidates<T>>` vec, you can use [decode_len()](https://paritytech.github.io/polkadot-sdk/master/frame_support/storage/types/struct.StorageValue.html#method.decode_len).

I'm not sure that it will help if `T::OnInactiveCollator::on_inactive_collator()` needs to modify this struct anyway, but it's probably a good optimization to do here anyway.